cmd/openshift-install/gather.go

```diff
@@ -197,3 +202,34 @@ func unSupportedPlatformGather(directory string) error {
 
 	return logGatherBootstrap(gatherBootstrapOpts.bootstrap, 22, gatherBootstrapOpts.masters, directory)
 }
+
+func logClusterOperatorConditions(ctx context.Context, config *rest.Config) error {
+	client, err := configclient.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "creating a config client")
+	}
+
+	operators, err := client.ConfigV1().ClusterOperators().List(metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "listing ClusterOperator objects")
+	}
+
+	for _, operator := range operators.Items {
+		logger := logrus.WithField("operator", operator.ObjectMeta.Name)
+		for _, condition := range operator.Status.Conditions {
+			if (condition.Type == configv1.OperatorAvailable || condition.Type == configv1.OperatorUpgradeable) && condition.Status == configv1.ConditionTrue {
+				continue
+			}
+			if (condition.Type == configv1.OperatorDegraded || condition.Type == configv1.OperatorProgressing) && condition.Status == configv1.ConditionFalse {
+				continue
+			}
+			logger.WithFields(logrus.Fields{
```

Comment:
I think a more human readable error message would be appropriate, in place of this fields based.
`Cluster operator NAME is not Available because REASON: MESSAGE`

Reply:
>     Cluster operator NAME is not Available because REASON: MESSAGE

Done with 7c7e4cf66 -> 618a84fd3.